WordPressUI/WordPressUI/FancyAlert/FancyButton.swift

```diff
@@ -8,61 +8,138 @@ open class FancyButton: UIButton {
+        didSet {
+            configureBackgrounds()
+        }
+    }
+    @objc public dynamic var secondaryNormalBorderColor = Secondary.normalBorderColor {
+        didSet {
+            configureBackgrounds()
+        }
+    }
+    @objc public dynamic var secondaryHighlightBackgroundColor = Secondary.highlightBackgroundColor {
+        didSet {
+            configureBackgrounds()
+        }
+    }
+    @objc public dynamic var secondaryHighlightBorderColor = Secondary.highlightBorderColor {
+        didSet {
+            configureBackgrounds()
+        }
+    }
 
     /// Style: Disabled State
     ///
-    @objc public dynamic var disabledBackgroundColor = Disabled.backgroundColor
-    @objc public dynamic var disabledBorderColor = Disabled.borderColor
+    @objc public dynamic var disabledBackgroundColor = Disabled.backgroundColor {
+        didSet {
+            configureBackgrounds()
+        }
+    }
+    @objc public dynamic var disabledBorderColor = Disabled.borderColor {
+        didSet {
+            configureBackgrounds()
+        }
+    }
 
     /// Style: Title!
     ///
-    @objc public dynamic var titleFont = Title.defaultFont
-    @objc public dynamic var primaryTitleColor = Title.primaryColor
-    @objc public dynamic var secondaryTitleColor = Title.secondaryColor
-    @objc public dynamic var disabledTitleColor = Title.disabledColor
+    @objc public dynamic var titleFont = Title.defaultFont {
+        didSet {
+            configureTitleLabel()
+        }
+    }
+    @objc public dynamic var primaryTitleColor = Title.primaryColor {
+        didSet {
+            configureTitleColors()
+        }
+    }
+    @objc public dynamic var secondaryTitleColor = Title.secondaryColor {
+        didSet {
+            configureTitleColors()
+        }
+    }
+    @objc public dynamic var disabledTitleColor = Title.disabledColor {
+        didSet {
+            configureTitleColors()
+        }
+    }
 
     /// Insets to be applied over the Contents.
     ///
-    @objc public dynamic var contentInsets = Metrics.contentInsets
+    @objc public dynamic var contentInsets = Metrics.contentInsets {
+        didSet {
+            configureInsets()
+        }
+    }
 
     /// Indicates if the current instance should be rendered with the "Primary" Style.
     ///
     @IBInspectable var isPrimary: Bool = false {
         didSet {
-            configureButton()
+            configureBackgrounds()
+            configureTitleColors()
         }
     }
 
 
     // MARK: - LifeCycle Methods
 
+    open override func didMoveToWindow() {
+        super.didMoveToWindow()
+        configureAllTheThings()
+    }
+
     open override func awakeFromNib() {
         super.awakeFromNib()
-        configureButton()
+        configureAllTheThings()
     }
 
 
-    /// Configure the appearance of the button.
+    /// Setup: Everything = [Insets, Backgrounds, titleColor(s), titleLabel]
+    ///
+    private func configureAllTheThings() {
```

Comment:
I love this, but could we change it to (much more boring) `configureAppearance`? :D

Reply:
LOL yeah i couldn't come up with something fancy, so... `all the things` sounded proper. Updated accordingly, thank you James!!